README.md

```diff
@@ -31,86 +25,186 @@ Templates in this repository include:
+*(don't include the "<>" symbols)*
+### Mac (darwin OS)
+
+#### amd64
+
+```sh
+curl -L -o /usr/local/bin/func "https://github.com/knative/func/releases/download/knative-v1.16.1/func_darwin_amd64"
+```
+
+#### arm64
+
+```sh
+curl -L -o /usr/local/bin/func "https://github.com/knative/func/releases/download/knative-v1.16.1/func_darwin_arm64"
+```
+
+*NOTE: After downloading on MacOS and Linux, you might need to make the file
+executable*
+
+```sh
+chmod +x /usr/local/bin/func
+```
+
+### Get a containerization technology
+These are some open-source examples of what you can use with Functions. You will
+need some tools to atleast build and push your images. This list is not exhaustive.
+
+[Buildah](https://github.com/containers/buildah/blob/main/install.md) - CLI tool
+to build your images.
+
+[Podman](https://podman.io/docs/installation#installing-on-linux) - complementary
+to Buildah. Helps you manage and modify your images. Uses Buildah's golang API.
+Can be installed independently. (*You can get this as a standalone tool*)
+
+[Docker Engine](https://docs.docker.com/engine/install/) - Helps you build and 
+containerize your applications. (*You can get this as a standalone tool*)
+
+#### Download local cluster runner (kind)
+Please refer to **kind**
+[installation page](https://kind.sigs.k8s.io/docs/user/quick-start/#installation)
+or download any other runner that you like.
+#### Download cli commands for k8s (kubectl)
+In order to interact with the objects in k8s, its recommended to get
+[kubectl](https://kubernetes.io/docs/tasks/tools/).
+
 ## How To Use
 You use these templates by creating your function via `--repository` flag which
-means "create my function with this template". 
+means "create my function with this template".
+
+Create your functions directory and `cd` into it
+
+```
+mkdir -p ~/testing/myfunc && cd ~/testing/myfunc
+```
 
-- Create a function with **hello template** in **golang** within the current directory
+- Create a function in **golang** with **hello template** within the new (current and empty) directory
 
 ```
-func create --repository=https://github.com/gauron99/func-templates -t=hello -l go
+func create --repository=https://github.com/gauron99/func-templates --language go --template=hello
 ```
 
 - TODO add more examples / explanations
-### Prerequisites
-- Download func
-- Download docker
-- Download local cluster runner (kind)
-- Download cli commands for k8s (kubectl)
 ### Build a Function
 
 #### Using the Host Builder
-This is the prefered way
+***NOTE**: Currently, this is under heavy construction and will become the default in the
+future.*
 
-#### Using Alternative Builders
-This is the old way (pack/s2i)
+In order to use the Host Builder in the mean time, you will need to enable it via a ENV variable
+and set it in `func` via CLI flag.
+
+Alternativelly, run this in your terminal, or put in your config file (*bashrc*)
+```bash
+export FUNC_ENABLE_HOST_BUILDER=1 FUNC_BUILDER=host FUNC_CONTAINER=false
```

Comment:
This works, but I would only specify ENABLE_HOST_BUILDER as an exported env var, and instead use --builder and --container as flags. This because env vars have very high precidence and may confuse users why the default flag value or func.yaml vale is ignored.